hashmap.js

```diff
@@ -26,9 +26,11 @@
 		switch (arguments.length) {
 			case 0: break;
 			case 1: {
-				if(Array.isArray(other)) {
-					this.from2DArray(other);
-				} else {
+				if ('length' in other) {
+					// Flatten 2D array to alternating key-value array
+					multi(this, 
+						Array.prototype.concat.apply([], other));

```

Comment:
Great man! You probably have or aim for a line length limit. But there's no case like this and I rather have a single line.
Please either merge the two or assign to a variable first, both are fine.

Looking good!